app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt

```diff
@@ -69,7 +69,7 @@ class LockboxAutofillService(
             .take(1)
             .subscribe { passwords ->
                 val possibleValues = passwords.filter {
-                    titleFromHostname(it.hostname).startsWith(requestingPackage, true)
+                    it.hostname.contains(requestingPackage, true)
```

Comment:
I think `contains` is too loose; it would succeed for `hostname = "https://www.facebook.com"` where `requestingPackage` is:

* "book.com"
* "book.co"
* "ebook.com"
* "ebook.co"
* "facebook.co"

I think the ultimate solution is to use the output from #375 and something that tightens up (and hopefully simplifies) doing an exact match or descendant sub-domains (e.g., `endsWith(".${privateDomain}")` but internationalized).

Until #375 is resolved, we can probably live with this, but not for Beta.  I can update #375 to include its use for sub-domain matching as above.

Reply:
let's move that thought into #375. I'm going to treat this ticket as covering as much view parsing for autofill-able sections as possible, and deal with the domain parsing there.